.github/workflows/nightly_build.yaml

```
@@ -14,11 +14,19 @@ jobs:
 
     permissions:
       contents: read
+      id-token: write
       packages: write
 
+    env:
+      COSIGN_EXPERIMENTAL: 1
+
     steps:
       - name: Checkout
         uses: actions/checkout@v3
```

a good practice is to use the git hash instead of the tag and have a comment to which release is pointing to. Dependabot when update the release can also update the comment :)

```suggestion
        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
```

same applies to the other actions

---

There is another PR adressing this practice already.